### ISSUE TITLE
Prevent blocking during placement

### ISSUE THREAD

**Issue by nvs** [nvs/gem#132]
This would assist players in knowing whether or not their latest placement in fact blocks the monsters' path. Now, some might say that not blocking is a skill one must learn. In a sense, that is a correct notion. However, that is probably the only reason to not implement automatic block checking. Typically one blocks due to a mistake or oversight. And often the results can be game ending.

Regardless, this will not be implemented without a bit of feedback since this would essentially change the core gameplay. Although, I personally see no honest downsides.

**Comment by nvs** [comment]
Another interesting notion is that if blocks were guaranteed not to happen, then I do not believe that creeps would need the ability to attack. In turn, this would mean that a player's maze being attacked would no longer be a concern.

**Comment by nvs** [comment]
Of course, that does not mean that creeps will not become confused. Because of this, issuing a move order periodically to them may be something to consider.

**Comment by nvs** [comment]
At the very least, an improvement to the existing block behavior could be implemented.

**Comment by nvs** [comment]
Implementation of this will proceed. Details are still up in the air, but due to the nature of the problem (a 2D grid where a path must be guaranteed between landmarks and predictable wall extensions) it may not be necessary to implement a pathfinding algorithm (e.g. A*).

**Comment by nvs** [comment]
This should be much simpler to introduced in Lua. Jass had the ops limit, and working around it was a pain. Not to mention the lack of performant data structures.

**Comment by GaDaHo** [comment]
It's a good check but I agree to all you said so do as you please is my gesture but somehow I use to get annoyed when you use to swap towers in previous versions when units are moving through my maze that causes the swap at that point to place the rock in the pathing of the unit that can cause maze block and then you know what happens and according to where it's blocked if thou seek to destroy rock it can lead to loss and either way if blocked the creeps destroy even tower to pass. Fun either way meaning is it not nice to hear your friend cry but not curse his maze is futile. But I see you have fixed this swapping issue so I'm happy but your choice for this "Prevent blocking during placement." might take away Focus if you know what I mean.
